### PR TITLE
[vcpkg] Add policy VCPKG_POLICY_SKIP_DUMPBIN_CHECKS

### PR DESCRIPTION
**Pull request by Neumann-A**
- also corrects some cmake advices. Paths should always be handled with "${PATH}" due to possible spaces

- allows users/ports to skip the dumpbin check if necessary, e.g. if dumpbin simply fails on the compiled dll or for other reasons

### CODE REVIEW

File: toolsrc/src/vcpkg/build.cpp [hunk 1]
```diff
@@ -173,6 +173,7 @@ namespace vcpkg::Build
     static const std::string NAME_EMPTY_INCLUDE_FOLDER = "PolicyEmptyIncludeFolder";
     static const std::string NAME_ALLOW_OBSOLETE_MSVCRT = "PolicyAllowObsoleteMsvcrt";
     static const std::string NAME_ALLOW_RESTRICTED_HEADERS = "PolicyAllowRestrictedHeaders";
+    static const std::string SKIP_DUMPBIN_CHECKS = "PolicySkipDumpbinChecks";
```
Comment (vicroms) [inline toolsrc/src/vcpkg/build.cpp]: Change to `NAME_SKIP_DUMPBIN_CHECKS`

**Review by vicroms — CHANGES_REQUESTED** [review]

File: toolsrc/src/vcpkg/postbuildlint.cpp [hunk 2]
```diff
@@ -349,8 +349,8 @@ namespace vcpkg::PostBuildLint
             const fs::path relative_path = found_file.string().erase(
                 0, current_buildtrees_dir.string().size() + 1); // The +1 is needed to remove the "/"
             System::printf(
-                "\n    file(COPY ${CURRENT_BUILDTREES_DIR}/%s DESTINATION ${CURRENT_PACKAGES_DIR}/share/%s)\n"
-                "    file(RENAME ${CURRENT_PACKAGES_DIR}/share/%s/%s ${CURRENT_PACKAGES_DIR}/share/%s/copyright)\n",
+                "\n    file(COPY \"${CURRENT_BUILDTREES_DIR}/%s\" DESTINATION \"${CURRENT_PACKAGES_DIR}/share/%s\")\n"
+                "    file(RENAME \"${CURRENT_PACKAGES_DIR}/share/%s/%s\" \"${CURRENT_PACKAGES_DIR}/share/%s/copyright\")\n",
```
Comment (ras0219-msft) [inline toolsrc/src/vcpkg/postbuildlint.cpp]: Since you're tweaking this anyway, it would be better to suggest using `configure_file()` with `COPYONLY`.

**Review by ras0219-msft — CHANGES_REQUESTED** [review]
LGTM once comments are addressed.

**Review by vicroms — APPROVED** [review]

**Comment by ras0219-msft** [comment]
LGTM, thanks!

**Pull request merged**